src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-sync-data-info/rgw-sync-data-info.component.spec.ts

```diff
@@ -22,4 +24,51 @@ describe('RgwSyncDataInfoComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should display "Up to Date" badge when zone is up to date', () => {
+    component.zone = {
+      timestamp: null
+    };
+    fixture.detectChanges();
+    const upToDateBadge = fixture.debugElement.query(By.css('.badge-success'));
+    expect(upToDateBadge).toBeTruthy();
+    expect(upToDateBadge.nativeElement.textContent).toEqual('Up to Date');
+  });
+
+  it('should display correct sync status and last synced time', fakeAsync(() => {
+    component.zone = { syncstatus: 'Syncing', timestamp: new Date(Date.now() - 10 * 60 * 1000) };
+    fixture.detectChanges();
+
+    const statusElement = fixture.debugElement.query(By.css('li b'));
+    expect(statusElement.nativeElement.textContent).toContain('Status:');
+
+    const lastSyncedElement = fixture.debugElement.query(By.css('li.mt-4.fw-bold'));
+    expect(lastSyncedElement.nativeElement.textContent).toContain('Last Synced:');
+    const lastSyncedTimestamp = fixture.debugElement.query(By.css('.badge-info'));
+    expect(lastSyncedTimestamp.nativeElement.textContent).toEqual('10 minutes ago');
+  }));
+
+  it('should display sync status in the popover', () => {
+    component.zone = {
+      syncstatus: 'Syncing',
+      timestamp: new Date(Date.now() - 10 * 60 * 1000),
+      fullSyncStatus: [
+        'full sync: 0/128 shards',
+        'incremental sync:128/128 shards',
+        'Data is behind on 31 shards'
+      ]
+    };
+    fixture.detectChanges();
+    const syncStatus = fixture.debugElement.query(By.css('.text-primary'));
+    expect(syncStatus).toBeTruthy();
+    expect(syncStatus.nativeElement.textContent).toEqual('Syncing');
+    const syncPopover = fixture.debugElement.query(By.css('a'));
+    syncPopover.triggerEventHandler('click', null);
+    fixture.detectChanges();
+    expect(syncPopover).toBeTruthy();
```

Comment:
```suggestion
    expect(syncStatus.nativeElement.textContent).toEqual('Syncing');
    const syncPopover = fixture.debugElement.query(By.css('a'));
    expect(syncPopover).toBeTruthy();
    syncPopover.triggerEventHandler('click', null);
    fixture.detectChanges();
```
Same thought as before on ` expect(syncStatus).toBeTruthy();`
If we still want to keep  ` expect(syncPopover).toBeTruthy();` then it better be before using `syncPopover`

Reply:
I this test we are checking if the sync status element is there and after that we check if the sync popover is there. Am not sure what's the confusion here?